datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -318,6 +318,19 @@ def create_scraper_configuration(self, instance=None):
         # Custom tags that will be sent with each metric
         config['custom_tags'] = instance.get('tags', [])
 
+        # Some tags can be ignored to reduce the cardinality.
+        # This can be useful for cost optimization in containerized environments when the openmetrics check is configured to collect custom metrics.
+        # Even when the Agent's Tagger is configured to add low-cardinality tags only, some tags can still generate unwanted metric contexts (e.g pod annotations as tags).
+        ignore_tags = instance.get('ignore_tags', default_instance.get('ignore_tags', []))
+        if ignore_tags:
+            ignored_tag_patterns = set()
+            for ignored_tag in ignore_tags:
+                ignored_tag_patterns.add(translate(ignored_tag))
+
+            if ignored_tag_patterns:
+                ignored_tags_re = compile('|'.join(ignored_tag_patterns))
+                config['custom_tags'] = [tag for tag in config['custom_tags'] if not ignored_tags_re.search(tag)]
```

Comment:
I may not be familiar enough with openmetrics to understand that part.
It looks to me that `custom_tags` is fetched from the config `tags` element.

And we then create a user-configurable filter to remove some values from `custom_tags`. What's the point of filtering a user-provided configuration with another user-provided configuration?

Reply:
Autodiscovery injects tagger tags using the `tags` field as well, that's what want to filter-out.

Reply:
I didn't realized that, thanks for clearing it out 👍

Reply:
no problem!